playbooks/adjust-cinder-backend.yml

```diff
@@ -0,0 +1,239 @@
+    notify:
+      - stop cinder services
+      - stop cinder backup service
+
+  - name: trigger stop all cinder services
+    meta: flush_handlers
+
+  - name: clear cinder services from db
+    command: mysql -e "DELETE cinder.services FROM cinder.services;"
+    when: set_cinder_host or update_cinder_backend
+    run_once: true
+    notify:
+      - restart cinder services
+      - restart cinder backup service
+
+  - name: update cinder volumes host entries in db
+    command: mysql -e "
+             DELETE cinder.services FROM cinder.services;
+             UPDATE cinder.volumes SET host='ceph@{{ cinder_backend_name }}' WHERE host is not NULL;
+             UPDATE cinder.backups SET host='ceph' WHERE host is not NULL;"
+    when: set_cinder_host or update_cinder_backend
+    run_once: true
+    notify:
+      - restart cinder services
+      - restart cinder backup service
+
+  - name: set cinder service host to ceph in cinder.conf
+    lineinfile: dest=/etc/cinder/cinder.conf
+                insertafter='^\[DEFAULT\]'
+                line='host = ceph'
+    when: set_cinder_host
+    notify:
+      - restart cinder services
+      - restart cinder backup service
+
+  - name: change cinder enabled_backends name in cinder.conf
+    replace:
+      dest: /etc/cinder/cinder.conf
+      regexp: "{{ item.regexp }}"
+      replace: "{{ item.replace }}"
+    with_items:
+      - { regexp: '^enabled_backends = {{ legacy_cinder_backend_name }}', replace: 'enabled_backends = {{ cinder_backend_name }}' }
+      - { regexp: '^\[{{ legacy_cinder_backend_name }}\]', replace: '[{{ cinder_backend_name }}]' }
+      - { regexp: '^volume_backend_name = {{ legacy_cinder_backend_name }}', replace: 'volume_backend_name = {{ cinder_backend_name }}' }
+    when: update_cinder_backend
+    notify:
+      - restart cinder services
+      - restart cinder backup service
+
+  - name: set default_volume_type to {{ cinder_volume_type }} in cinder.conf
+    lineinfile: dest=/etc/cinder/cinder.conf
+                insertafter='^\[DEFAULT\]'
+                line='default_volume_type = {{ cinder_volume_type }}'
+    when: set_default_volume_type
+    notify:
+      - restart cinder services
+      - restart cinder backup service
+
+  - name: trigger restart cinder services
+    meta: flush_handlers
+
+  - name: create cinder volume type {{ cinder_volume_type }}
+    cinder_volume_type:
+      volume_type: "{{ cinder_volume_type }}"
+      auth_url: "{{ endpoints.auth_uri }}"
+      login_username: admin
+      login_password: "{{ secrets.admin_password }}"
+      login_tenant_name: admin
+      extra_specs: "volume_backend_name={{ cinder_backend_name }}"
+    when: set_default_volume_type
+    run_once: true
+
+  - name: Get default volume_type id
+    shell: . /root/stackrc; openstack volume type show {{ cinder_volume_type }} -c id -f value
+    when: set_default_volume_type
+    run_once: true
+    register: default_volume_type_id
```

Comment:
Did you check the output of `cinder_volume_type`? That may have returned the UUID of the type, obviating the need for this task here.